### DISCUSSION TITLE
Is there something like a Response Interceptor in Adonis?

### DISCUSSION THREAD

**Discussion by LeCoupa** [adonisjs/core#1946] [Help] [answered]
I know about response macros but is there something like a global interceptor where all responses will go through in Adonis? For instance, to camelCase all keys in the object sent.

**Comment by RomainLanz** [comment]
Hey @LeCoupa! 👋 

You can create a downstream middleware to do this.

**Reply by aayush123** [reply]
Is there any mention of this in the docs that I can refer?

**Reply by RomainLanz** [reply, marked as answer]
You can look at the documentation about [middleware](https://docs.adonisjs.com/guides/middleware).
Write your code **after** the `await next()` call to execute anything after your handler's code.
